packages/firestore/src/local/remote_document_change_buffer.ts

```diff
@@ -99,6 +107,32 @@ export abstract class RemoteDocumentChangeBuffer {
     }
   }
 
+  /**
+   * Looks up an entry in the cache. The buffered changes will first be checked,
+   * and if no buffered change applies, this will forward to
+   * `RemoteDocumentCache.getEntry()`.
+   *
+   * @param transaction The transaction in which to perform any persistence
+   *     operations.
+   * @param documentKey The key of the entry to look up.
+   * @return The cached Document or NoDocument entry, or null if we have nothing
+   * cached.
+   */
+  getEntries(
+    transaction: PersistenceTransaction,
+    documentKeys: DocumentKeySet
+  ): PersistencePromise<NullableMaybeDocumentMap> {
+    const changes = this.assertChanges();
+
+    // Record the size of everything we load from the cache so we can compute a delta later.
```

Comment:
Question: I could add code to look up the keys in the buffer. I decided to postpone it because it will necessitate some logic to merge the buffer lookup results with results from cache, so I'd like to check beforehand if you think this optimization is worthwhile.